Dockerfile

```diff
@@ -222,38 +224,33 @@ RUN sed -i 's|http://archive.ubuntu.com|http://us.archive.ubuntu.com|g' /etc/apt
         mv "$SONAR_DIR"/build-wrapper-* "$SONAR_DIR"/build-wrapper/ && \
         # FIXME: The following symlink will no longer be required once images
         # that depend on carma-base change from wait-for-it.sh to wait-for-it
-        ln -s /usr/bin/wait-for-it /usr/bin/wait-for-it.sh
-
-USER carma
-
-# Install non-ros1 dependant version of catkin
-# This can be used without issue for ROS2 builds wheras the noetic version has compatability issues
-# install catkin_pkg
-RUN cd $HOME && \
-        mkdir catkin_ros2_ws && \
-        cd catkin_ros2_ws && \
-        git clone https://github.com/ros-infrastructure/catkin_pkg.git && \
-        cd catkin_pkg && \
+        ln -s /usr/bin/wait-for-it /usr/bin/wait-for-it.sh && \
+        # Install non-ros1 dependant version of catkin
+        # This can be used without issue for ROS2 builds wheras the noetic version has compatability issues
+        # install catkin_pkg
+        sudo -u carma mkdir /home/carma/catkin_ros2_ws && \
+        sudo -u carma git clone https://github.com/ros-infrastructure/catkin_pkg.git /home/carma/catkin_ros2_ws/catkin_pkg && \
+        cd /home/carma/catkin_ros2_ws/catkin_pkg && \
         # Checkout a known working commit
-        git checkout 60096f4b4a0975774651122b7e2d346545f8098a && \
-        python3 setup.py install --prefix $HOME/catkin --single-version-externally-managed --record foo --install-layout deb && \
-        cd ../ && \
+        sudo -u carma git checkout 60096f4b4a0975774651122b7e2d346545f8098a && \
+        sudo -u carma python3 setup.py install --prefix /home/carma/catkin --single-version-externally-managed --record foo --install-layout deb && \
         # install catkin
-        git clone https://github.com/ros/catkin.git && \
-        cd catkin && \
+        sudo -u carma git clone https://github.com/ros/catkin.git /home/carma/catkin_ros2_ws/catkin && \
```

Comment:
Is there a particular rationale for doing `sudo -u carma ...` here, as opposed to moving the `USER carma` line on 255 to somewhere above this block and then doing `sudo  ...` for the commands that need root? It seems like more of these commands need to be run as carma than as root anyway, so it'd be a bit easier to follow that way I think.